src/wrapper/jnienv.rs

```diff
@@ -1497,6 +1497,88 @@ impl<'a> JNIEnv<'a> {
         res
     }
 
+    /// Get a static field without checking the provided type against the actual
+    /// field.
+    #[allow(unused_unsafe)]
+    pub unsafe fn get_static_field_unsafe<T, U>(
+        &self,
+        class: T,
+        field: U,
+        ty: JavaType,
+    ) -> Result<JValue>
+    where
+        T: Desc<'a, JClass<'a>>,
+        U: Desc<'a, JFieldID<'a>>,
+    {
+        let class = class.lookup(self)?.into_inner();
+
+        let field_id = field.lookup(self)?.into_inner();
+
+        // TODO clean this up
```

Comment:
Why TODO?

Reply:
Uhm... I copied it from `get_field_unsafe`, with most of the code.

Maybe it refers to the differences between the `match ty { ... }` of `get_field_unsafe` and of `set_field_unsafe`.

Reply:
I see, thank you. I'd ask the maintainer then if it shall remain for any prospective refactoring.

Reply:
Yeah, I couldn't think of a good solution to all of the repetition at the time the original was written, hence the TODO. This is fine for now, TODO included. I'll take a look at it when I get around to doing the cleanup.